src/lib/stark/singAmmFundingOrder.js

```diff
@@ -0,0 +1,19 @@
+const P = require('aigle')
+const createSignedTransaction = require('../dvf/createSignedTransaction')
+
+module.exports = dvf => async data => ({
+  ...data,
+  starkPublicKey: await dvf.dvfStarkProvider.getPublicKey(),
+  // Need to do this in series since createSignedTransaction might call
+  // into ledger and these calls cannot be executed in parallel.
+  orders: await P.mapSeries(
+    data.orders,
+    async order => ({
+      ...order,
+      starkOrder: order.starkOrder.signature
+        // Only sign if there is no signature
```

Comment:
Can we elaborate in what contexts that could happen ?

Reply:
if someone used their own code to create and sign the `starkOrder`  they we wouldn't want to overwrite this signature. I guess you might ask why would they call `signAmmFundingOrder`? They might not call it directly but it called [here](https://github.com/DeversiFi/dvf-client-js/blob/bba3f9f49865cfe56a72b8f516231b5433754c08/src/api/postAmmFundingOrder.js#L48). But I think I might refactor things a bit and move the conditional out of this function

Reply:
fixed in 430423a5ab0512d79315c8efac8b481fc67977cb